### PR TITLE
Added JSDoc comments for the gulpfile.

### PR DESCRIPTION
**Pull request by jaboc83**

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -298,7 +361,13 @@ function thirdPartyBootstrap(watch, shouldMinify) {
       });
 }
 
-
+/**
+ * Compile a javascript file
+ * @param {string} srcDir Path to the src directory
+ * @param {string} srcFilename Name of the JS source file
+ * @param {string} destDir Destination folder for output script
+ * @param {Object} options
```
Comment (erwinmombay) [inline gulpfile.js]: `{!Object}`

File: gulpfile.js [hunk 2]
```diff
@@ -105,7 +125,12 @@ function compile(watch, shouldMinify) {
   thirdPartyBootstrap(watch, shouldMinify);
 }
 
-
+/**
+ * Compile all the css and drop in the build folder
+ *
+ * @return {!Promise} containing a Readable
+ *	Stream
```
Comment (erwinmombay) [inline gulpfile.js]: no need to for newline here i think

File: gulpfile.js [hunk 3]
```diff
@@ -187,6 +223,18 @@ function buildExtension(name, version, hasCss, options) {
...
+ * Build the JavaScript for the extension specified
+ *
+ * @param {string} js JavaScript file content
+ * @param {string} path Path to the extensions directory
+ * @param {string} name Name of the extension. Must be the sub directory in
+ *     the extensions directory and the name of the JS and optional CSS file.
+ * @param {string} version Version of the extension. Must be identical to
+ *     the sub directory inside the extension directory
+ * @param {!Object} options
+ * @return {!Object} Gulp object
```
Comment (erwinmombay) [inline gulpfile.js]: `{!Stream}` i think

File: gulpfile.js (hunk 1 above)
Comment (jaboc83) [inline gulpfile.js]: Looks like it handles nulls according to line 372?
```javascript
options = options || {};
```

Comment (erwinmombay) [inline gulpfile.js]: good catch. then please mark with `{?Object}` the `?` marks it as `nullable`

Comment (jaboc83) [inline gulpfile.js]: Did not know that one :)  Will do.

File: gulpfile.js (hunk 3 above)
Comment (jaboc83) [inline gulpfile.js]: Done.

File: gulpfile.js (hunk 2 above)
Comment (jaboc83) [inline gulpfile.js]: Done.

**Comment by erwinmombay** [comment]
LGTM

**Comment by erwinmombay** [comment]
@jaboc83 you'll need to rebase

**Comment by jaboc83** [comment]
Hoping I did that right...

**Comment by erwinmombay** [comment]
@jaboc83 yep! its green

**Pull request merged**